access/handler.go

```diff
@@ -9,20 +9,31 @@ import (
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
 	"github.com/onflow/flow-go/engine/common/rpc/convert"
 	"github.com/onflow/flow-go/model/flow"
 )
 
 type Handler struct {
-	api   API
-	chain flow.Chain
+	api            API
+	chain          flow.Chain
+	sgnIdcsDecoder hotstuff.BlockSignerDecoder
 }
 
-func NewHandler(api API, chain flow.Chain) *Handler {
-	return &Handler{
-		api:   api,
-		chain: chain,
+// HandlerOption is used to hand over optional constructor parameters
+type HandlerOption func(*Handler)
+
+func NewHandler(api API, chain flow.Chain, options ...HandlerOption) *Handler {
```

Comment:
used the 'optional-argument' pattern here, to avoid breaking the emulator: https://github.com/onflow/flow-emulator/blob/945d217ad3b30f8579aeed915a1c3345922bd56f/server/grpc.go#L54